SDK v2 migration guide.md

```diff
@@ -284,8 +284,9 @@ These span across all service clients.
 #### Notable breaking changes
 
 - Operations that offer concurrency protection using `ETag`s, now take a parameter `onlyIfUnchanged` that relies on the ETag property of the submitted entity.
-- `IotHubServiceClient.Query.CreateAsync<T>(...)` is now async.
```

Comment:
This function is no longer async since it just builds the query (the first service request isn't made until the iteration begins) so I'm stripping the ```async``` suffix here and elsewhere